src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -57,21 +58,29 @@ public Engine(PowerFxConfig powerFxConfig)
 
         // By default, we pull the core functions. 
         // These can be overridden. 
-        internal IEnumerable<TexlFunction> Functions => CreateResolverInternal().Functions;
+        internal TexlFunctionSet Functions => CreateResolverInternal().Functions;
 
         /// <summary>
         /// Get all functions from the config and symbol tables. 
         /// </summary>
-        public IEnumerable<FunctionInfo> FunctionInfos => Functions.Select(f => new FunctionInfo(f));
+        [Obsolete("Slow API. Use With* APIs to select functions.")]
```

Comment:
> [Obsolete("Slow API. Use With* APIs to select functions.")] [](http://example.com/codeflow?start=7&length=60)

These can't be Obsolete. 

These are public surface functions. (With* is not). We can't obsolete them until we have a suitable replacement. 

Ability to allow a host to enumerate all functions is still a valid scenario; we just don't want them to be doing it accidentally when they don't need to.